integrations/jira/main.py

```diff
@@ -35,46 +43,61 @@ async def setup_application() -> None:
 
-    async for projects in client.get_paginated_projects():
-        logger.info(f"Received project batch with {len(projects)} issues")
+    async for projects in client.get_all_projects():
+        logger.info(f"Received project batch with {len(projects)} projects")
         yield projects
 
 
 @ocean.on_resync(ObjectKind.ISSUE)
 async def on_resync_issues(kind: str) -> ASYNC_GENERATOR_RESYNC_TYPE:
-    client = JiraClient(
-        ocean.integration_config["jira_host"],
-        ocean.integration_config["atlassian_user_email"],
-        ocean.integration_config["atlassian_user_token"],
-    )
+    client = initialize_client()
+    config = typing.cast(JiraIssueResourceConfig, event.resource_config).selector
+    params = {}
+    if config.jql:
+        params["jql"] = config.jql
 
-    async for issues in client.get_paginated_issues():
+    if config.fields:
+        params["fields"] = config.fields
+
+    async for issues in client.get_all_issues(params):
         logger.info(f"Received issue batch with {len(issues)} issues")
         yield issues
 
 
 @ocean.router.post("/webhook")
 async def handle_webhook_request(data: dict[str, Any]) -> dict[str, Any]:
-    client = JiraClient(
-        ocean.integration_config["jira_host"],
-        ocean.integration_config["atlassian_user_email"],
-        ocean.integration_config["atlassian_user_token"],
-    )
-    logger.info(f'Received webhook event of type: {data.get("webhookEvent")}')
-    if "project" in data:
+    client = initialize_client()
+    webhook_event: str = data.get("webhookEvent", "")
+    logger.info(f"Received webhook event of type: {webhook_event}")
+    ocean_action = None
+    delete_action = False
+
+    if webhook_event in DELETE_WEBHOOK_EVENTS:
+        ocean_action = ocean.unregister_raw
+        delete_action = True
+    elif webhook_event in CREATE_UPDATE_WEBHOOK_EVENTS:
+        ocean_action = ocean.register_raw
+
+    if not ocean_action:
+        logger.info("Webhook event not recognized")
+        return {"ok": True}
+
+    if "project" in webhook_event:
         logger.info(f'Received webhook event for project: {data["project"]["key"]}')
-        project = await client.get_single_project(data["project"]["key"])
-        await ocean.register_raw(ObjectKind.PROJECT, [project])
-    elif "issue" in data:
+        if delete_action:
+            project = data["project"]
+        else:
+            project = await client.get_single_project(data["project"]["key"])
+        await ocean_action(ObjectKind.PROJECT, [project])
+    elif "issue" in webhook_event:
         logger.info(f'Received webhook event for issue: {data["issue"]["key"]}')
-        issue = await client.get_single_issue(data["issue"]["key"])
-        await ocean.register_raw(ObjectKind.ISSUE, [issue])
+        if delete_action:
+            issue = data["issue"]
+        else:
+            issue = await client.get_single_issue(data["issue"]["key"])
+        await ocean_action(ObjectKind.ISSUE, [issue])
```

Comment:
I think we should perform a request to get the issue with the `fields` as well, so we don't override it with null once we receive an event. 
Maybe `JQL` as well? as we might receive an event of an issue that we are not interested which in regular `resync` wouldn't been inserted, but in real time event it would've.

Reply:
All fields are returned by default, so we shouldn't have null overwrites